webrender/src/render_backend.rs

```diff
@@ -400,6 +400,7 @@ pub struct RenderBackend {
     recorder: Option<Box<ApiRecordingReceiver>>,
     sampler: Option<Box<AsyncPropertySampler + Send>>,
     size_of_op: Option<VoidPtrToSizeFn>,
+    namespace_alloc_by_client: bool,
```

Comment:
is there any reason not to *always* allocate it on the client side?

Reply:
If client always allocate namespace id(32bit), the id allocation is not easy when RenderApiSender::create_api() is called from several different processes.

In gecko's use case, create_api() is always called by single process, then the id allocation is easy by client side.